src/toil/provisioners/aws/awsProvisioner.py

```diff
@@ -1121,96 +1225,90 @@ def _getBoto3BlockDeviceMappings(cls, type_info: InstanceType, rootVolSize: int
+                rightState = rightState or i['State']['Name'] == 'stopping' or i['State']['Name'] == 'stopped'
             return rightType and rightState
 
         return [i for i in all_instances if instanceFilter(i)]
 
-    def _filter_nodes_in_cluster(self, instance_type: Optional[str] = None, preemptible: bool = False) -> List[Boto2Instance]:
-        """
-        Get Boto2 instance objects for the nodes in the cluster filtered by preemptability.
-        """
-
-        instances = self._get_nodes_in_cluster(instance_type, include_stopped_nodes=False)
-
-        if preemptible:
-            return [i for i in instances if i.spot_instance_request_id is not None]
-
-        return [i for i in instances if i.spot_instance_request_id is None]
-
     def _getSpotRequestIDs(self) -> List[str]:
         """
         Get the IDs of all spot requests associated with the cluster.
         """
 
         # Grab the connection we need to use for this operation.
-        ec2 = self.aws.boto2(self._region, 'ec2')
+        ec2: EC2Client = self.aws.client(self._region, 'ec2')
 
-        requests = ec2.get_all_spot_instance_requests()
-        tags = ec2.get_all_tags({'tag:': {_TAG_KEY_TOIL_CLUSTER_NAME: self.clusterName}})
-        idsToCancel = [tag.id for tag in tags]
-        return [request for request in requests if request.id in idsToCancel]
+        requests: List[SpotInstanceRequestTypeDef] = ec2.describe_spot_instance_requests()["SpotInstanceRequests"]
+        tag_filter: FilterTypeDef = {"Name": "tag:" + _TAG_KEY_TOIL_CLUSTER_NAME, "Values": [self.clusterName]}
+        tags: List[TagDescriptionTypeDef] = ec2.describe_tags(Filters=[tag_filter])["Tags"]
+        # tags = ec2.get_all_tags({'tag:': {_TAG_KEY_TOIL_CLUSTER_NAME: self.clusterName}})
+        idsToCancel = [tag["ResourceId"] for tag in tags]
+        return [request["SpotInstanceRequestId"] for request in requests if request["InstanceId"] in idsToCancel]
 
     def _createSecurityGroups(self) -> List[str]:
         """
         Create security groups for the cluster. Returns a list of their IDs.
         """
+        def group_not_found(e: ClientError) -> bool:
+            retry = (get_error_status(e) == 400 and 'does not exist in default VPC' in get_error_body(e))
+            return retry
 
         # Grab the connection we need to use for this operation.
         # The VPC connection can do anything the EC2 one can do, but also look at subnets.
-        vpc = self.aws.boto2(self._region, 'vpc')
+        boto3_ec2: EC2Client = self.aws.client(region=self._region, service_name="ec2")
 
-        def groupNotFound(e):
-            retry = (e.status == 400 and 'does not exist in default VPC' in e.body)
-            return retry
-        # Security groups need to belong to the same VPC as the leader. If we
-        # put the leader in a particular non-default subnet, it may be in a
-        # particular non-default VPC, which we need to know about.
```

Comment:
Isn't this still true or important?

Reply:
I think this still applies as the code is just a port, so I can add the comment back